src/main.rs

```diff
@@ -52,47 +52,73 @@ fn read_input<T: AsRef<str>>(filename: T) -> std::io::Result<(PathBuf, String)>
     }
 }
 
-fn process_arguments(size: TerminalSize, args: Arguments) -> Result<(), Box<dyn Error>> {
+fn process_file(
+    filename: &str,
+    size: TerminalSize,
+    args: &Arguments,
+) -> Result<(), Box<dyn Error>> {
+    let (base_dir, input) = read_input(filename)?;
+    let mut options = Options::empty();
+    options.insert(Options::ENABLE_TASKLISTS);
+    options.insert(Options::ENABLE_STRIKETHROUGH);
+    let parser = Parser::new_ext(&input, options);
+
+    if args.dump_events {
+        mdcat::dump_events(&mut std::io::stdout(), parser)?;
+    } else {
+        let syntax_set = SyntaxSet::load_defaults_newlines();
+        mdcat::push_tty(
+            &mut stdout(),
+            &args.terminal_capabilities,
+            TerminalSize {
+                width: args.columns,
+                ..size
+            },
+            parser,
+            &base_dir,
+            args.resource_access,
+            syntax_set,
+        )?;
+    }
+    Ok(())
+}
+
+fn process_arguments(size: TerminalSize, args: Arguments) -> Result<(), String> {
```

Comment:
I guess `process_arguments` doesn't need to return a `Result` anymore, now that we handle errors directly.

Why not make it return the intended exit code, and use `std::process::exit(process_arguments(…))` in `main()`?

Reply:
I thought about it, but I was trying to make the least amount of code changes that might not be necessary.   I will make this change though.

Reply:
Technically, you do not need process arguments if you put all that functionality from process_arguments into main.

Reply:
Inlining `process_arguments` to `main` is a good idea, now that `process_file` is separate :+1: